File: AD-HOC/3058/main.dart
import 'dart:io';

void main() {
  List<double> prices = [];
  int n = int.parse(stdin.readLineSync()!);
  for (int store = 0; store < n; store++) {
    var [p, g] = stdin.readLineSync()!.split(' ').map(double.parse).toList();
    prices.add(1000 * p / g);
  }
  prices.sort();
  print('${prices.first.toStringAsFixed(2)}');
}

File: AD-HOC/3058/main.js
let input = require('fs').readFileSync('/dev/stdin', 'utf-8');
let lines = input.split('\n');
let prices = [];
let n = parseInt(lines[0]);
for (let i = 1; i <= n; i++) {
    var [p, g] = lines[i].split(' ').map(Number);
    prices.push(1000 * p / g);
}
prices.sort((a, b) => a - b);
console.log(`${prices[0].toFixed(2)}`);
